tests/accelerators/test_tpu_backend.py

```diff
@@ -186,3 +190,77 @@ def test_set_devices_if_none_tpu():
 
     trainer = Trainer(accelerator="tpu", tpu_cores=8)
     assert trainer.devices == 8
+
+
+@RunIf(tpu=True)
+def test_manual_optimization_tpus(tmpdir):
```

Comment:
I feel like we already have several tests that check the layers during manual optimization. Can't we parametrize an existing one?

Reply:
Parametrize what? We have kept the tests separate for TPUs. Also, for the reason you mentioned, I have added only one test to cover it all.

Reply:
as a middle ground, we could also try to reuse an existing ManualOpt model from the test_manual_optimization file

Reply:
@awaelchli Haha, I did add that in this [commit](https://github.com/PyTorchLightning/pytorch-lightning/pull/8458/commits/0f66a539ca2d3f079b5a3a91c413aa568106872a), but later decided it would be better to add more assertions and functionalities.